libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx

```diff
@@ -4,28 +4,56 @@
  */
 
 import React, {
+  forwardRef,
+  useRef,
   MouseEvent,
   ReactElement,
   useCallback,
   useMemo,
   useState,
 } from "react"
-import { APIFC, BaseGenericComponent } from "generic-view/utils"
 import styled, { css } from "styled-components"
+import { BaseGenericComponent } from "generic-view/utils"
+import { TooltipPlace } from "device/models"
 
-export const Tooltip: APIFC & {
+export const Tooltip: BaseGenericComponent<
+  undefined,
+  undefined,
+  { place?: TooltipPlace }
+> & {
   Anchor: typeof TooltipAnchor
   Content: typeof TooltipContent
-} = ({ children }) => {
+} = ({ children, place = "bottom-right" }) => {
   const [anchorPosition, setAnchorPosition] = useState<{
     top?: number
     left?: number
   }>({})
 
-  const handleAnchorHover = useCallback((event: MouseEvent) => {
-    const rect = event.currentTarget.getBoundingClientRect()
-    setAnchorPosition({ top: rect.top + rect.height, left: rect.left })
-  }, [])
+  const contentRef = useRef<HTMLDivElement>(null)
+
+  const handleAnchorHover = useCallback(
+    (event: MouseEvent) => {
+      const anchorRect = event.currentTarget.getBoundingClientRect()
+      const contentReact = contentRef.current?.getBoundingClientRect()
+
+      if (contentReact === undefined) {
+        return
+      }
+
+      if (place === "bottom-right") {
+        const top = anchorRect.top + anchorRect.height
+        const left = anchorRect.left
+
+        setAnchorPosition({ left, top })
+      } else if (place === "bottom-left") {
+        const top = anchorRect.top + anchorRect.height
+        const left = anchorRect.left - contentReact.width + anchorRect.width
+
+        setAnchorPosition({ left, top })
+      }
```

Comment:
I’d prefer to keep the if statements because there might be more positions added in the future, and it makes it easier to read and understand the logic of positioning based on the placement. This way, each position will have clearly defined logic, which improves code readability.